fix(django): ensure person DELETE is scoped by team_id in _raw_delete_batch

eli-r-ph opened the pull request:
## Problem

Now that `posthog_persons` is partitioned by `team_id` we need to ensure team ID is included in filtering clauses in relevant queries against the table, to ensure the database doesn't fan out to every partition.

<!-- Who are we building for, what are their needs, why is this important? -->

<!-- Does this fix an issue? Uncomment the line below with the issue ID to automatically close it when merged -->
<!-- Closes #ISSUE_ID -->

## Changes
* Fixes `DELETE` stmt generated in Team model `util.py`'s `_raw_delete_batch`
* Add query snapshot to test case to verify the change has the desired effect
<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

## How did you test this code?
Locally and in CI
<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changelog: (features only) Is this feature complete?
No update required
<!-- Yes if this is okay to go in the changelog. No if it's still hidden behind a feature flag, or part of a feature that's not complete yet, etc.  -->
<!-- Removing this section does not mean the changelog bot won't pick it up, because *some people* like to not use the template, so we can't rely on it existing. -->

eli-r-ph committed:
ensure posthog_person DELETE stmt is scoped by team_id in _raw_delete_batch team utils

eli-r-ph committed:
update test and check in new snapshot

eli-r-ph committed:
refactor the DELETE query so that (id, team_id) are tightly scoped in each _raw_delete_batch

github-actions[bot] committed:
Update query snapshots

eli-r-ph commented on posthog/api/test/__snapshots__/test_team.ambr:
```diff
@@ -0,0 +1,42 @@
+         "team_id") IN ((174,
+                         120),(175,
+                               120))
+  '''
```
☝️ the addition of `team_id` in this deletion on `posthog_persondistinctid` concerns me a bit as there's no composite index on `(id, team_id)` on that table, just separate `id` and `team_id` indices...

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

webjunkie reviewed (approved):
Seems like that does it!
The snapshots seem flaky though. I'll try to address that.

webjunkie committed:
fix: handle nested tuple IN clauses in test snapshot normalization

The snapshot normalization regex wasn't handling the nested tuple format
used in the new DELETE queries: WHERE (id, team_id) IN ((1, 2), (3, 4)).
This caused snapshots to flip-flop with actual database IDs instead of
being normalized to deterministic values.

Added regex pattern to normalize `IN ((1, 2), (3, 4))` to `IN ((1, 2) /* ... */)`.

github-actions[bot] committed:
Update query snapshots
